src/metabase/models/secret.clj

```diff
@@ -25,6 +27,46 @@
 ;;; ---------------------------------------------- Hydration / Util Fns ----------------------------------------------
 ;; none yet
 
+(def
+  ^{:doc "The attributes of a secret which, if changed, will result in a version bump" :private true}
+  bump-version-keys
+  [:kind :source :value])
+
+(defn latest-for-id
+  "Returns the latest Secret instance for the given `id` (meaning the one with the highest `version`)."
+  {:added "0.41.0"}
+  [id]
+  (db/select-one Secret :id id {:order-by [[:version :desc]]}))
+
+(defn upsert-secret-value!
```

Comment:
What's the plan for deletion of secrets/secret versions?

Scenario 1 - I added a data source and decided to remove it from Metabase. I expect the password to be deleted from the DB as well.

Scenario 2 - If I did not enable encryption at first but then later turn it on, I expect the old unencrypted value to be deleted from the DB.

Reply:
Good points.  I will address in a separate PR to keep these fairly bite sized.